File: app/src/main/java/com/test/xyz/demo/presentation/projectdetails/di/RepoDetailsFragmentComponent.java
package com.test.xyz.demo.presentation.projectdetails.di;

import com.test.xyz.demo.presentation.common.di.scope.ActivityScope;
import com.test.xyz.demo.presentation.projectdetails.ProjectDetailsFragment;

import dagger.Subcomponent;

@ActivityScope
@Subcomponent(
        modules = {RepoDetailsFragmentModule.class}
)
public interface RepoDetailsFragmentComponent {
    void inject(ProjectDetailsFragment projectDetailsFragment);
}

File: app/src/main/kotlin/com/test/xyz/demo/presentation/projectdetails/di/RepoDetailsFragmentComponent.kt
package com.test.xyz.demo.presentation.projectdetails.di

import com.test.xyz.demo.presentation.common.di.scope.ActivityScope
import com.test.xyz.demo.presentation.projectdetails.ProjectDetailsFragment

import dagger.Subcomponent

@ActivityScope
@Subcomponent(modules = arrayOf(RepoDetailsFragmentModule::class))
interface RepoDetailsFragmentComponent {
    fun inject(projectDetailsFragment: ProjectDetailsFragment)
}

